bftengine/src/bftengine/ClientsManager.cpp

```diff
@@ -265,7 +266,7 @@ bool ClientsManager::noPendingAndRequestCanBecomePending(NodeIdType clientId, Re
   uint16_t idx = clientIdToIndex_.at(clientId);
   const ClientInfo& c = indexToClientInfo_.at(idx);
 
-  if (c.currentPendingRequest != 0) return false;  // if has pending request
+  if (c.currentPendingRequest != 0 || c.waitsForExecution == true) return false;  // if has pending request
```

Comment:
// if has pending request, or the request was committed but not yet executed

Reply:
I Will fix this in my next PT